tests/integration/base.py

```diff
@@ -34,6 +40,56 @@ def __exit__(self, *args, **kwargs):
         pass
 
 
+class ZipFile(zipfile.ZipFile):
+    def extract(self, member, path=None, pwd=None):
+        if not isinstance(member, zipfile.ZipInfo):
+            member = self.getinfo(member)
+
+        if path is None:
+            path = os.getcwd()
+
+        ret_val = self._extract_member(member, path, pwd)
+        attr = member.external_attr >> 16
+        os.chmod(ret_val, attr)
+        return ret_val
+
+
+class GraknDistribution(object):
+    DISTRIBUTION_LOCATION = 'external/graknlabs_grakn_core/grakn-core-all-mac.zip'
+    DISTRIBUTION_ROOT_DIR = 'grakn-core-all-mac'
+
+    def __init__(self):
+        self.__unpacked_dir = None
+
+    def _unpack(self):
+        self.__unpacked_dir = tempfile.mkdtemp(prefix='grakn')
+        with ZipFile(GraknDistribution.DISTRIBUTION_LOCATION) as zf:
+            zf.extractall(self.__unpacked_dir)
+
+    @property
+    def __grakn_root_directory(self):
+        return os.path.join(self.__unpacked_dir, GraknDistribution.DISTRIBUTION_ROOT_DIR)
+
+    def __enter__(self):
```

Comment:
As we've discussed over Slack, let's combine `start` with `__enter__` and `stop` with `__exit__`. `__enter__` and `__exit__` are just one liners so there isn't much sense in splitting them